openhands-sdk/openhands/sdk/context/skills/skill.py

```diff
@@ -366,3 +368,128 @@ def load_user_skills() -> list[Skill]:
         f"Loaded {len(all_skills)} user skills: {[s.name for s in all_skills]}"
     )
     return all_skills
+
+
+# Public skills repository configuration
+PUBLIC_SKILLS_REPO = "https://github.com/OpenHands/skills"
+PUBLIC_SKILLS_BRANCH = "main"
+
+
+def load_public_skills(
+    repo_url: str = PUBLIC_SKILLS_REPO,
+    branch: str = PUBLIC_SKILLS_BRANCH,
+) -> list[Skill]:
+    """Load skills from the public OpenHands skills repository.
+
+    This function fetches skills from the public skills registry at
+    https://github.com/OpenHands/skills. Skills are downloaded and loaded
+    dynamically, allowing users to get the latest skills without SDK updates.
+
+    Args:
+        repo_url: URL of the skills repository. Defaults to the official
+            OpenHands skills repository.
+        branch: Branch name to load skills from. Defaults to 'main'.
+
+    Returns:
+        List of Skill objects loaded from the public repository.
+        Returns empty list if loading fails.
+
+    Example:
+        >>> from openhands.sdk.context import AgentContext
+        >>> from openhands.sdk.context.skills import load_public_skills
+        >>>
+        >>> # Load public skills
+        >>> public_skills = load_public_skills()
+        >>>
+        >>> # Use with AgentContext
+        >>> context = AgentContext(skills=public_skills)
+    """
+    all_skills = []
+
+    # Extract owner and repo name from GitHub URL
+    try:
+        # Support both HTTPS and SSH URLs
+        if repo_url.startswith("https://github.com/"):
+            parts = repo_url.replace("https://github.com/", "").rstrip("/").split("/")
+        elif repo_url.startswith("git@github.com:"):
+            parts = repo_url.replace("git@github.com:", "").rstrip("/").split("/")
+        else:
+            logger.warning(
+                f"Invalid repository URL format: {repo_url}. "
+                f"Expected GitHub URL format."
+            )
+            return all_skills
+
+        if len(parts) < 2:
+            logger.warning(f"Invalid repository URL format: {repo_url}")
+            return all_skills
+
+        owner, repo = parts[0], parts[1].replace(".git", "")
+
+        # Use GitHub API to list files in the skills/ subdirectory
+        api_url = f"https://api.github.com/repos/{owner}/{repo}/git/trees/{branch}:skills?recursive=1"
```

Comment:
@OpenHands, any more efficient way to get this? Maybe just clone the skills repo under `.openhands` folder?